backend/api/visualization.proto

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option go_package = "github.com/kubeflow/pipelines/backend/api/go_client";
+package api;
+
+import "google/api/annotations.proto";
+import "backend/api/error.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  responses: {
+    key: "default";
+    value: {
+      schema: {
+        json_schema: {
+          ref: ".api.Status";
+        }
+      }
+    }
+  }
+  // Use bearer token for authorizing access to job service.
+  // Kubernetes client library(https://kubernetes.io/docs/reference/using-api/client-libraries/)
+  // uses bearer token as default for authorization. The section below
+  // ensures security definition object is generated in the swagger definition.
+  // For more details see https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#securityDefinitionsObject
+  security_definitions: {
+    security: {
+      key: "Bearer";
+      value: {
+        type: TYPE_API_KEY;
+        in: IN_HEADER;
+        name: "authorization";
+      }
+    }
+  }
+  security: {
+    security_requirement: {
+      key: "Bearer";
+      value: {};
+    }
+  }
+};
+
+service VisualizationService {
+  rpc CreateVisualization(CreateVisualizationRequest) returns (Visualization) {
+    option (google.api.http) = {
+      post: "/apis/v1beta1/visualizations"
+      body: "visualizaiton"
+    };
+  }
+}
+
+// Create visualization by providing the type of visualization that is desired
+// and input data paths. Input dat paths are assumed to be unique and are used
+// for determining output path.
+message CreateVisualizationRequest {
+  Visualization visualization = 1;
+}
+
+message Visualization {
+  // Output. Unique visualization ID. Generated by API server.
+  string id = 1;
+
+  // Type of visualization to be generated.
+  // This is required when creating the pipeline through CreateVisualization
+  // API.
+  enum Type {
+    ROC_CURVE = 0;
+  };
+  Type type = 2;
+
+  // Array of paths of data to be used during generation of visualizations.
+  // This is required when creating the pipeline through CreateVisualization
+  // API.
+  repeated string inputPaths = 3;
+
+  // Output. Path of visualization within artifact store.
+  string path = 4;
```

Comment:
as discussed with @neuromage offline, what do you think if it returns the html instead? 
that way the frontend would not need to know the detail of how html is stored.

Reply:
That is totally possible. The only reason for this implementation is to mimic the current metadata structure. Where a metadata file is fetched and the provided file path within the metadata is then fetched to render visualizations. Either option is possible but removing the responsibility from the frontend might be best.

Reply:
storing html in artifact store is for caching purpose which might be an implementation detail. Imaging start without having cache, but instead the API does conversion each time, this interface might not work well. Returning HTML might be a better option. WDYT?

Reply:
I agree that returning HTML would be the optimal solution. It would reduce network calls on within the frontend and also allow for caching without breaking changes in the future.